internal/executor/reporter/job_event_reporter.go

```diff
@@ -34,9 +33,17 @@ type JobEventReporter struct {
 
 	jobRunStateStore *job.JobRunStateStore
 	clusterContext   clusterContext.ClusterContext
+	clock            clock.Clock
+	maxBatchSize     int
 }
 
-func NewJobEventReporter(clusterContext clusterContext.ClusterContext, jobRunState *job.JobRunStateStore, eventSender EventSender) (*JobEventReporter, chan bool) {
+func NewJobEventReporter(
+	clusterContext clusterContext.ClusterContext,
+	jobRunState *job.JobRunStateStore,
+	eventSender EventSender,
+	clock clock.Clock,
```

Comment:
I'd remove the clock parameter here.  The only reason you'll need to set it is for unit tests and for that you are probably in the same package so you can just override the field directly.

Reply:
Sounds sensible, but I'd need to swap it to have a Run func as the clock gets used in the constructor.